src/models/client-sdk.ts

```diff
@@ -0,0 +1,26 @@
+import { AdditionalHttpHeaders, ClientSDKHeaderValues } from '@yext/answers-core';
+
+/**
+ * Allows users to specify additional values for specific HTTP headers.
+ *
+ * @public
+ */
+export interface CustomHttpHeaders extends AdditionalHttpHeaders {
```

Comment:
can we alias the original AdditionalHttpHeaders as something else like `import { AdditionalHttpHeaders as blah }` and then export a new interface called AdditionalHttpHeaders? I'm not sure if the export-star eslint plugin yells at us for that but it could be hard to come up with a new name every time we add a new library to this "additional http header stack" with every new lib we add on top

Reply:
though, we have answers-core-re-exports so maybe it won't complain after all?

Reply:
since we can just remove it from there

Reply:
I thought about doing that first, but the problem is that we're exporting everything from Core in Headless, so when API extractor builds the docs for those, it would resolve any inheritDoc references to `AdditionalHttpHeaders` as the new one we specify in Headless. we can rename each place we import it from Core (including answers-core-re-exports) but we wouldn't be able to change references in the docs

Reply:
@nmanu1, I think that behavior sounds okay to me? If you're looking at the docs for `AdditionalHttpHeaders` in the Headless Docs, I would expect to see the model which includes `ANSWERS_HEADLESS`.

Reply:
Ok, in that case, I think most of the time it would be fine. The only place where it might be odd is if we use Core's `AdditionalHttpHeaders` as a type somewhere in Headless (such as in the AnswersHeadless constructor). Even though we rename it something else when we import it, I think the docs still point to `AdditionalHttpHeaders`, which would now resolve to the interface defined in Headless. That would be slightly inaccurate for the linked type, but if we're ok with that, I can make the change

Reply:
Why are we using Core's `AdditionalHttpHeaders` in the `AnswersHeadless` constructor?

Reply:
The AnswersHeadless constructor stores the exact headers that would be sent to Core, which includes the Headless agent. This would not be allowed if we used the interface defined in Headless as the type (which doesn't allow the Headless agent to be specified)

Reply:
In that case, maybe it would be simpler to stick with the current approach and perhaps consider some renaming? `CustomHttpHeaders` could become `HeadlessAdditionalHttpHeaders` and  `CustomClientSDKHeaderValues` could become `ClientSDKHeaderValuesExcludingHeadless`

Reply:
Sure, I can rename. That would give us a consistent way to name this in other libraries if we build on top of this

Reply:
In my experience, renaming can cause confusion about which interface is actually being used. I prefer additional interfaces with slightly longer names if it increases clarity and provides us with accurate documentation. I also considered removing the static type checking to ensure ANSWERS_HEADLESS isn't supplied and simply relying on runtime checks, however in the past we have decided to use more complicated types for the sake of type checking at build time (e.g. supporting apiKey or auth token but not both), and I think we should make that same decision here as well.